kmod/src/dir.c

```diff
@@ -1111,46 +1111,55 @@ static void *scoutfs_follow_link(struct dentry *dentry, struct nameidata *nd)
 				   corrupt_symlink_missing_item,
 				   "ino %llu size %llu", scoutfs_ino(inode),
 				   size);
+		kfree(path);
 		ret = -EIO;
 
 	} else if (ret == 0 && path[size - 1]) {
 		scoutfs_corruption(sb, SC_SYMLINK_NOT_NULL_TERM,
 				   corrupt_symlink_not_null_term,
 				   "ino %llu last %u",
 				   scoutfs_ino(inode), path[size - 1]);
+		kfree(path);
 		ret = -EIO;
 	}
 
 out:
-	if (ret < 0) {
-		kfree(path);
```

Comment:
This now leaks the allocated path on errors that aren't caught by the two cases that call _corruption().  I'm not sure why the free was removed and put in those two cases, but undoing that change will fix it.